### PR TITLE
feat(process): redo bootstrap after X amount of time

### LINKED ISSUE #20: Automatically restart replication every week
To get a downloads data that is always up to date, we need to restart the replication from seq=0 every month.

There should be ways on heroku to do that.

### PR DESCRIPTION
**Pull request by Haroenv**
fixes #20

In the 'watch' step, I get the state after each iteration, then if now - last time bootstrap was done is more than the time to redo the bootstrap, the settings

```js
stateManager.set({
  seq: 0,
  bootstrapDone: false,
});
```

are being set, which *should* make the bootstrap start over.

I am not very sure how to test this locally without having to wait a long time...

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -179,6 +180,17 @@ function watch({ seq }) {
...
         )
+        .then(stateManager.get)
+        .then(({ bootstrapLastDone }) => {
+          const now = new Date();
+          const lastBootstrapped = new Date(bootstrapLastDone);
+          if (now - lastBootstrapped > c.timeToRedoBootstrap) {
+            stateManager.set({
+              seq: 0,
+              bootstrapDone: false,
+            });
```
Comment (Haroenv) [inline index.js]: I’m doubting if I need `.then(()=>exit(0))` is needed here.

Comment (redox) [inline index.js]: What do you think @vvo?

File: index.js [hunk 2]
```diff
@@ -104,6 +104,7 @@ function bootstrap(state) {
           log.info('⛷ Bootstrap: done');
           return stateManager.save({
             bootstrapDone: true,
+            bootstrapLastDone: new Date().toISOString(),
```
Comment (vvo) [inline index.js]: Isn't Date.now() sufficient (store as a timestamp)?

File: index.js (hunk 1 above)
Comment (vvo) [inline index.js]: Where would you want to put that exit(0)?

Comment (vvo) [inline index.js]: Where would you want to put that exit(0)?

**Pull request merged**